google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -795,6 +799,7 @@ public TransactionRunner readWriteTransaction() {
     @Override
     public void close() {
       synchronized (lock) {
+        numSessionsReleased++;
```

Comment:
I don't think releasing sessions is the same as closing the session pool. Releasing is to do with the user handing the session back to the pool. Closing is shutting down the connection all together.

I think you want to add this to `releaseSession` on line 1522.